snuba/stateful_consumer/states/catching_up.py

```diff
@@ -13,13 +17,29 @@ class CatchingUpState(State[ConsumerStateCompletionEvent, ConsumerStateData]):
     consumption.
     """
 
+    def __init__(
+        self,
+        consumer_builder: ConsumerBuilder
+    ) -> None:
+        super(CatchingUpState, self).__init__()
+        self.__consumer_builder = consumer_builder
+        self.__consumer = None
+
     def signal_shutdown(self) -> None:
-        pass
+        if self.__consumer:
+            self.__consumer.signal_shutdown()
 
     def handle(self, state_data: ConsumerStateData) -> Tuple[ConsumerStateCompletionEvent, ConsumerStateData]:
-        # TODO: Actually consume cdc topic while discarding xids that were
-        # already in the dump
+        assert state_data.snapshot_id is not None
+        assert state_data.transaction_data is not None
+
+        self.__consumer = self.__consumer_builder.build_snapshot_aware_consumer(
+            snapshot_id=state_data.snapshot_id,
+            transaction_data=state_data.transaction_data,
```

Comment:
Why isn't `build_snapshot_aware_consumer` just reusing the same `ConsumerStateData` data type as it's parameter?

Reply:
because build_snapshot_aware_consumer came before I introduced CatchingUpState and I did not think about that when I added CatchingUpState. Will change.